cpp/test/sg/fil_test.cu

```diff
@@ -301,15 +415,18 @@ class BaseFilTest : public testing::TestWithParam<FilTestParams> {
   {
     // predict on host
     std::vector<float> want_preds_h(ps.num_preds_outputs());
-    std::vector<float> want_proba_h(ps.num_proba_outputs());
+    want_proba_h.resize(ps.num_proba_outputs());
     int num_nodes = tree_num_nodes();
+    auto infer    = [&](int root, int row) {
```

Comment:
I don't think it makes the code more readable.

Reply:
will revert then